queryviews/models.py

```diff
@@ -0,0 +1,94 @@
+from django.contrib.postgres.fields import JSONField
+from django.core.serializers.json import DjangoJSONEncoder
+from django.apps import apps
+from django.db import models
+
+from matview.dbutils import get_matview_name
+
+
+queries = {}
+
+
+def register_query(model, filters):
+    name = get_matview_name(model, filters)
+    app_label, model_name = model.split('.')
+
+    if filters:
+        class Meta:
+            managed = False
+            db_table = name
+        attrs = {
+            'ncid': models.CharField('ncid', max_length=12),
+            'data': JSONField(encoder=DjangoJSONEncoder),
+            'Meta': Meta,
+            '__module__': 'queryviews.models',
+            'filters': filters,
+        }
+
+        # Instansiate a Model class with our generated name and attributes
+        query_model = type(name, (models.Model,), attrs)
+        # Register our generated model in our lookup registry
+        queries.setdefault(app_label, {}).setdefault(model_name, {})[name] = query_model
+
+    class Meta:
+        managed = False
+        db_table = name + '__count'
+    attrs = {
+        'count': models.IntegerField(),
+        'Meta': Meta,
+        '__module__': 'queryviews.models',
+    }
+    # Instansiate a Model class with our generated name and attributes
+    count_model = type(name + '_count', (models.Model,), attrs)
```

Comment:
This must be the first time I've seen this usage of ``type`` other than deep in some esoteric library.

Reply:
is that good or bad

Reply:
good I guess